app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedViewModel.kt

```diff
@@ -0,0 +1,162 @@
+package org.oppia.android.app.home.recentlyplayed
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import org.oppia.android.R
+import org.oppia.android.app.fragment.FragmentScope
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.app.model.PromotedActivityList
+import org.oppia.android.app.model.PromotedStory
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.domain.topic.TopicListController
+import org.oppia.android.domain.translation.TranslationController
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import org.oppia.android.util.parser.html.StoryHtmlParserEntityType
+import javax.inject.Inject
+
+/** View model for [RecentlyPlayedFragment]. */
+@FragmentScope
+class RecentlyPlayedViewModel @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val fragment: Fragment,
+  private val topicListController: TopicListController,
+  @StoryHtmlParserEntityType private val entityType: String,
+  private val resourceHandler: AppLanguageResourceHandler,
+  private val translationController: TranslationController,
+) {
+
+  private var internalProfileId: Int = -1
+
+  /**
+   * Warning: it is required to call [setInternalProfileId] before accessing this property.
```

Comment:
The KDoc should explain what this is first, before any such warnings.

Reply:
Done